portfolio/src/components/Connect.vue

```diff
@@ -258,14 +289,74 @@ export default {
                 this.alertMessage = "Comments delete successfully"
 
                 // Call the get command to update the UI with the comment sent via the POST above
-                await this.getDataFromServer()
+                let respData = await this.getDataFromServer()
+                this.updateCommentList(respData)
             } else {
                 this.alertType = "error"
                 this.alertMessage = "You are not in the sudoers file, this incident will be reported"
             }
 
             this.alert = true
-        }        
+        },
+
+        commentSuccessCallback: async function () {
+            let respData = await this.getDataFromServer()
+            console.log(respData)
+            this.updateCommentList(respData)
+        },
+
+        changeNumComments: async function() {
+
+            // Reset last page value since there will be a new number of comments per page
+            this.lastPage = Number.MAX_SAFE_INTEGER            
+
+            let respData = await this.getDataFromServer()
+
+            if (respData.length < 1){
+                // Set the last page as the current page, as there are no comments in the current page
+                this.lastPage = this.page - 1
+
+                // If the current page is not 1, then show user the previous page. If it is 1 it means there are no comments at all, so do nothing
+                this.page != 1 && (this.page -= 1)
+
+                let respData = await this.getDataFromServer()
+                
+                this.updateCommentList(respData)
+                
+                return
+            }
+
+            this.updateCommentList(respData)
+        },
+
+        changeCommentPage: async function (isForward) {
+            if (isForward) {
+                if (this.page < this.lastPage){this.page += 1}                
+            }
+
+            else {
+                if (this.page > 1){ this.page -= 1 }                
+            }
+            
+            // Get data for new comments page
+            let respData = await this.getDataFromServer()
```

Comment:
for readability: would be better to pass in page number as parameter to getDataFromServer(), it's clearer that getDataFromServer() supports pagination rather than getting everything

Reply:
So I should pass in this.numComments and this.page, or would it be better to just rename the function to something like getPaginatedData?

Reply:
rename sounds good! I think passing parameter also has benefit of making this class stateless. It's not super clear which page is going to be fetched when calling the function.